File: backend/src/ERPBPM.Domain/Entities/Project.cs
using ERPBPM.Domain.Common;

namespace ERPBPM.Domain.Entities;

public class Project : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }

    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }

    public string Status { get; set; } = "Active";

    // tambahan
    public int Progress { get; set; } = 0;
    public decimal Budget { get; set; } = 0;

    // relasi
    public Guid OwnerId { get; set; }
    public User Owner { get; set; } = null!;

    public Guid? ManagerId { get; set; }
    public User? Manager { get; set; }

    public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
    public ICollection<ProjectMember> Members { get; set; } = new List<ProjectMember>();
}

File: frontend-mobile/lib/features/projects/domain/entities/project.dart
import 'package:equatable/equatable.dart';

class Project extends Equatable {
  final String id; // Changed from int to String for GUID
  final String name;
  final String description;
  final DateTime startDate;
  final DateTime endDate;
  final String status;
  final int progress;
  final double budget;
  final String? managerName;

  const Project({
    required this.id,
    required this.name,
    required this.description,
    required this.startDate,
    required this.endDate,
    required this.status,
    required this.progress,
    required this.budget,
    this.managerName,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        startDate,
        endDate,
        status,
        progress,
        budget,
        managerName,
      ];
}
